src/frontends/tensorflow_common/src/op/hsv_to_rgb.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/split.hpp"
+#include "utils.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_hsv_to_rgb_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"HSVToRGB"});
+    auto images = node.get_input(0);
+    auto node_name = node.get_name();
+
+    auto const_minus_one_i = make_shared<v0::Constant>(element::i32, Shape{}, -1);
+    auto channels = make_shared<v1::Split>(images, const_minus_one_i, 3);
+
+    auto hh = channels->output(0).get_node_shared_ptr();
+    auto ss = channels->output(1).get_node_shared_ptr();
+    auto vv = channels->output(2).get_node_shared_ptr();
+
+    auto new_images = hsv_to_rgb(hh, ss, vv);
```

Comment:
I think you need to change a bit `hsv_to_rgb`. It should expect `ov::Output<ov::Node>` instead of shared_ptr<Node> since in your case Node will be the same for all channels that is `Split` node. So the current `hsv_to_rgb` takes the same zero index port data `h` and it leads to mismatch inference results in layer tests

Reply:
> I think you need to change a bit `hsv_to_rgb`. It should expect `ov::Output<ov::Node>` instead of shared_ptr Node since in your case Node will be the same for all channels that is `Split` node. So the current `hsv_to_rgb` takes the same zero index port data `h` and it leads to mismatch inference results in layer tests

That makes sense. So to clarify, I should refactor the hsv_to_rgb function to take an `ov::Output<ov::Node>`? Or should I overload it?

Reply:
refactor, just change shaerd_prt<Node> to ov::Output<ov::Node>